prow/config/config.go

```diff
@@ -503,6 +543,58 @@ func parseConfig(c *Config) error {
 
 	return nil
 }
+
+func (c *Config) decorationRequested() bool {
+	for _, vs := range c.Presubmits {
+		for i := range vs {
+			if vs[i].Decorate {
+				return true
+			}
+		}
+	}
+
+	for _, js := range c.Postsubmits {
+		for i := range js {
+			if js[i].Decorate {
+				return true
+			}
+		}
+	}
+
+	for i := range c.Periodics {
+		if c.Periodics[i].Decorate {
+			return true
+		}
+	}
+
+	return false
+}
+
+func setDecorationDefaults(provided, defaults *kube.DecorationConfig) *kube.DecorationConfig {
+	merged := &kube.DecorationConfig{}
+	if provided != nil {
+		merged = provided
+	}
+
+	if merged.Timeout == 0 {
```

Comment:
Would it make sense to make these pointers so that we can distinguish 0 from unset?

Reply:
That implies that we care about an explicitly set value of 0. In this context, that value makes no sense so keeping it as-is seems fine.